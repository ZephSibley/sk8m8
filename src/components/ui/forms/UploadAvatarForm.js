import React, { useState } from 'react';
import { Formik, Field } from 'formik';
import *  as Yup from 'yup';
import FormHelperText from '@material-ui/core/FormHelperText';
import Button from '@material-ui/core/Button';

import formStyles from '../../../styles/forms';
import ImageInput from '../inputs/ImageInput';


const UploadAvatarForm = ({ requests }) => {
    // prop: requests; http client

    const [submitError, setSubmitError] = useState(null);

    const validationSchema = Yup.object().shape({
        avatar: Yup.mixed()
            .required("Please select an image")
            .test(
                "fileSize",
                "Your image is too big :(",
                value => value.size <= 104857600
            )
            .test(
                "fileType",
                "png or jpeg only please",
                value => value.type === 'image/jpeg' || value.type === 'image/png'
            )
    })

    const formClasses = formStyles()
    return (
        <Formik
            initialValues={{
                avatar: null,
            }}
            validationSchema={validationSchema}
            onSubmit={(values, { setSubmitting }) => {
                setSubmitting(true);
                requests.post(
                    `${process.env.REACT_APP_ENDPOINT}account/updateavatar`,
                    values,
                    { withCredentials: true }
                ).catch(err => {
                    setSubmitError(err.message);
                    setSubmitting(false);
                })
            }}

            render={({
                values,
                errors,
                touched,
                handleChange,
                setFieldValue,
                handleBlur,
                handleSubmit,
                isSubmitting,
                isValid,
            }) => (
                    <form
                        onSubmit={handleSubmit}
                        className={formClasses.form}
                    >
                        <Field
                            name="avatar"
                            component={ImageInput}
                            title="Avatar"
                            setFieldValue={setFieldValue}
                            errorMessage={errors.marker_video ?
                                errors.marker_video : ''
                            }
                            touched={touched.marker_video}
                            onBlur={handleBlur}
                        />
                        {values.avatar ? 
                            <Button
                                variant="contained"
                                color="primary"
                                component="button"
                                type="submit"
                                disabled={isSubmitting || !isValid}
                            >
                                Upload
                            </Button>
                        : ''
                        }
                        {submitError ?
                            <FormHelperText
                                    component={'span'}
                                    error={true}
                                >
                                    {submitError}
                            </FormHelperText>
                        : ''
                        }
                    </form>
                )}
        />
    )
}

export default UploadAvatarForm;