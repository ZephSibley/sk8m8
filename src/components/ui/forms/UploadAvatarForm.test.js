import React from 'react';
import renderer from 'react-test-renderer';
import UploadAvatarForm from './UploadAvatarForm';

test('Basic Rendering', () => {
    const component = renderer.create(
        <UploadAvatarForm requests={{}} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})