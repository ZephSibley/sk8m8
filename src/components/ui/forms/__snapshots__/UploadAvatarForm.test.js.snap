// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<form
  className="makeStyles-form-1"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root"
  >
    <input
      accept="image/png,image/jpeg"
      id="image-upload"
      name="avatar"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <label
      htmlFor="image-upload"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Avatar
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
            />
          </svg>
        </span>
      </span>
    </label>
  </div>
  
  
</form>
`;
